README.md

```diff
@@ -0,0 +1,91 @@
+
+We have added a failpoint `PushDownTestSwitcher` in the function `canFuncBePushed` to hijack our customized push down condition.
+And we will push down all functions in our integration tests using `export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"`, see the `run_tests.sh`[./push-down-test/run-tests.sh].
+
+## How to add test cases?
+
+- Test cases location
+    
+    All test cases should be placed in `push-down-test/sql` directory, and all SQL files will be executed in lexical order by file name.
+    So if your case want to prepare some data before running tests, you can add some suffix to the file name, eg: `xxx.1.sql` and `xxx.2.sql`.
+    
+- Test case file name convention
+
+    All test cases should have the suffix `.sql` as file extension. But you can add a `xxx.md` documentation to explain the complicated cases.
+    
+- Test case example
+
+    ```sql
+    use push_down_test_db;
+    create table tb2
+    (
+    date datetime,
+    date_2 datetime,
+    date_3 datetime
+    );
+    insert into tb2 (date, date_2, date_3)
+    values ('1-1-1:10:10', '1-2-1:10:10', '1-2-1:10:10');
+    select *
+    from tb2;
+    SELECT CAST(date AS time), cast(date_2 as date), cast(date_3 as date)
+    from tb2;
+    ```
+
+## How to run it in local environment?
+
+Because of the `copr-test` is the integration test for TiKV Coprocessor, so there is a minimal requirement if you want
+run the integration test in your local environment.
+
+### Minimal requirement components
+
+- PD: [PD](https://github.com/pingcap/pd) is the abbreviation for Placement Driver. It is used to manage and schedule the TiKV cluster.
+
+    1. Make sure ​Go​ (version 1.12) is installed.
+    2. Use make to install PD. PD is installed in the bin directory.
+
+- TiKV: [Building and setting up a development workspace](https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#building-and-setting-up-a-development-workspace)
+- TiDB: [Building TiDB on a local OS/shell environment](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md#building-tidb-on-a-local-osshell-environment)
+
+The `copr-test` integration test will retrieve the source code of the TiDB master branch automatically (but you can specify the
+TiDB source code path by `$tidb_src_dir` instead of using remote latest TiDB source code), so you should only prepare
+the build environment for it. The binaries of TiKV and PD should be provided by environment variables (`$pd_bin` and `$tikv_bin`).
+
+### Example guide
+
+```shell
+mkdir ~/devel/opensource/
+cd ~/devel/opensource
+git clone https://github.com/pingcap/pd.git
+git clone https://github.com/tikv/copr-test.git
+git clone https://github.com/tikv/tikv.git
+
+cd ~/devel/opensource/pd
+make
+
+cd ~/devel/opensource/tikv
+make
+
+cd ~/devel/opensource/copr-test
+pd_bin=~/devel/opensource/pd/bin/pd-server tikv_bin=~/devel/opensource/tikv/target/release/tikv-server make
```

Comment:
`make push-down-test` is better. because we may add new test in this repo

Reply:
I think use shell option (such as `-d`, `-p`, `-k`, etc ) to specify the tikv_bin, pd_bin is better

Reply:
I think to use the environment variables is better (we can export the related environment variables in our localhost instead of passing the command-line parameters every time).